### PR TITLE
Add purescript formatting support

### PR DESCRIPTION
**Pull request by ajnsit**
This PR adds support for formatting purescript sources using [purs-tidy](https://github.com/natefaubion/purescript-tidy).

### CODE REVIEW

File: layers/+lang/purescript/funcs.el [hunk 1]
```diff
@@ -37,3 +37,46 @@
...
+(defun spacemacs/purescript-purs-tidy-format-buffer ()
+  "Format buffer with purs-tidy."
+  (interactive)
+  (if (executable-find "purs-tidy")
+      (let*  ((extension (file-name-extension (or buffer-file-name "tmp.purs") t))
+              (tmpfile (make-temp-file "~fmt-tmp" nil extension))
+              (coding-system-for-read 'utf-8)
+              (coding-system-for-write 'utf-8)
+              (outputbuf (get-buffer-create "*~fmt-tmp.purs*")))
+        (unwind-protect
```
Comment (lebensterben) [inline layers/+lang/purescript/funcs.el:60]: For this `unwind-protect` form:

First, there's no need for `progn` in for the ELSE part in a `if` form:
```elisp
(progn
  (message "Formatting failed!")
  (display-buffer outputbuf)))
```

Second, it seems that
```elisp
(progn
  (delete-file tmpfil)))))
```
is intend to be the clean up forms should the body form `unwind-protect` fails. But it's now in the body form, due to the missing parenthesis after `(display-buffer outputbuf)))`.

**Review by lebensterben — CHANGES_REQUESTED** [review]

File: layers/+lang/purescript/funcs.el (hunk 1 above)
Comment (ajnsit) [reply layers/+lang/purescript/funcs.el:60]: You are right about the unwind-protect, the temporary file should always be deleted on cleanup. Fixed now.

Regarding the progn, I find it clearer to group the statements for else together, however I'm okay either way. I have removed the extra progn.

**Review by lebensterben — APPROVED** [review]

**Comment by lebensterben** [comment]
Please update CHANGELOG.develop before I merge it.

**Review by lebensterben — APPROVED** [review]

**Comment by ajnsit** [comment]
> Please update CHANGELOG.develop before I merge it.

Done!

**Comment by lebensterben** [comment]
Merged, thanks for your contributions.

**Pull request merged**